components/Valuation/Graph.tsx

```diff
@@ -0,0 +1,125 @@
+import React, { useEffect, useState, useRef } from "react";
+import { Metaverse } from "../../lib/enums";
+import { createChart, UTCTimestamp } from "lightweight-charts";
+import { getValuationDailyData } from "../../lib/FirebaseUtilities";
+
+const Graph = ({ metaverse }: Metaverse) => {
+  const metaverseGraph = useRef<HTMLDivElement>(null);
+  const [values, setValues] = useState([]);
+  const [symbol, setSymbol] = useState<"ETH" | "USDC" | "Metaverse">("ETH");
+
+  useEffect(() => {
+    (async () => setValues(await getValuationDailyData(metaverse)))();
+  },[metaverse]);
+
+  useEffect(() => {
+    const chart = createChart(metaverseGraph?.current!, {
+      width: metaverseGraph?.current!.clientWidth,
+      height: 197,
+      timeScale: {
+        fixLeftEdge: true,
+        fixRightEdge: true,
+        timeVisible: true,
+      },
+      rightPriceScale: {
+        scaleMargins: {
+          top: 0.3,
+          bottom: 0.25,
+        },
+        borderVisible: false,
+      },
+      layout: {
+        backgroundColor: "#131722",
+        textColor: "#d1d4dc",
+      },
+      grid: {
+        vertLines: {
+          color: "rgba(42, 46, 57, 0)",
+        },
+        horzLines: {
+          color: "rgba(42, 46, 57, 0.6)",
+        },
+      },
+    });
+    const areaSeries = chart.addAreaSeries({
+      topColor: "rgba(38,198,218, 0.56)",
+      bottomColor: "rgba(38,198,218, 0.04)",
+      lineColor: "rgba(38,198,218, 1)",
+      lineWidth: 2,
+      title:"Daily Volume"
+    });
+
+    const volumeSeries = chart.addHistogramSeries({
+      color: "#26a69a",
+      title:"Floor Price",
+      priceFormat: {
+        type: "volume",
+      },
+      priceScaleId: "",
+      scaleMargins: {
+        top: 0.8,
+        bottom: 0,
+      },
+    });
+    areaSeries.setData(
+      values.map((data) => {
+        return {
+          time: new Date(data.time) / 1000,
+          value:
+            data.dailyVolume[
+              (symbol === "ETH" && "ethPrediction") ||
+                (symbol === "USDC" && "usdPrediction") ||
+                (symbol === "Metaverse" && "metaversePrediction")
+            ],
+        };
+      })
+    );
+    volumeSeries.setData(
+      values.map((data) => {
+        return {
+          time: new Date(data.time) / 1000,
```

Comment:
There's some type issues with this. Also coming from the missing type of values, so try to work on that and lets see what happens. When possible try to fix all type issues before submiting work, but im also here to help.